### PR TITLE
Make PhysicalFile Watcher and Provider disposable

### PR DESCRIPTION
**Pull request by BrennanConroy**
#132

### CODE REVIEW

**Comment by BrennanConroy** [comment]
Also, removed PhysicalFilesWatcher from PhysicalFileInfo because it wasn't used anywhere and not accessible to anyone.

**Comment by BrennanConroy** [comment]
@Tratcher @dougbu

File: test/Microsoft.AspNet.FileProviders.Physical.Tests/PhysicalFileProviderTests.cs [hunk 1]
```diff
@@ -20,14 +20,16 @@ public class PhysicalFileProviderTests
...
         {
-            var provider = new PhysicalFileProvider(Directory.GetCurrentDirectory());
-            var info = provider.GetFileInfo("File.txt");
-            Assert.NotNull(info);
-            Assert.True(info.Exists);
-
-            info = provider.GetFileInfo("/File.txt");
-            Assert.NotNull(info);
-            Assert.True(info.Exists);
+            using (var provider = new PhysicalFileProvider(Directory.GetCurrentDirectory()))
```
Comment (dougbu) [inline test/Microsoft.AspNet.FileProviders.Physical.Tests/PhysicalFileProviderTests.cs]: Seems like all the changes from here on are about reducing file system burden while testing i.e. releasing resources ASAP. If there's an exception to that, let me know.

**Comment by dougbu** [comment]
:shipit:

Do we have `Hosting`, `Mvc` and `StaticFiles` follow-ups planned to add `using`s, now that it's useful?

**Comment by BrennanConroy** [comment]
Not really, I took a quick look at places this is used and I can change a few tests to dispose but can't easily dispose the other ones.

**Pull request merged**